specs/reject.go

```diff
@@ -0,0 +1,214 @@
+func NewRejectService(logger *slog.Logger, googleClient *google.Google, db *gorm.DB, config RejectConfig) *RejectService {
+	return &RejectService{
+		Logger:       logger.With("component", "specs_reject"),
+		GoogleClient: googleClient,
+		DB:           db,
+		Config:       config,
+	}
+}
+
+// findStaleSpecs identifies specifications that have "Drafting" or "Braindump" status
+// and have not been updated in the last 6 months
+func (r *RejectService) findStaleSpecs() ([]*db.Spec, error) {
+	var specs []*db.Spec
+	err := r.DB.
+		Where("LOWER(status) IN ?", []string{"drafting", "braindump"}).
+		Where("google_doc_updated_at < ?", time.Now().AddDate(0, -6, 0)).
+		Find(&specs).Error
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to query stale specs: %w", err)
+	}
+
+	return specs, nil
+}
+
+// RejectAllStaleSpecs finds and rejects all stale specifications
+func (r *RejectService) RejectAllStaleSpecs(ctx context.Context) error {
+	r.Logger.Info("starting stale spec rejection job",
+		"dry_run", r.Config.DryRun,
+	)
+	r.FailedCount = 0
+	r.RejectedCount = 0
+	cleanupID := uuid.New().String()
+
+	specs, err := r.findStaleSpecs()
+	if err != nil {
+		return fmt.Errorf("failed to find stale specs: %w", err)
+	}
+	if len(specs) == 0 {
+		r.Logger.Info("no stale specs to reject")
+		return nil
+	}
+
+	// Process specs sequentially
+	for _, spec := range specs {
+		select {
+		case <-ctx.Done():
+			r.Logger.Info("rejection job cancelled")
+			return ctx.Err()
+		default:
+		}
+
+		logger := r.Logger.With("spec_id", spec.ID, "doc_id", spec.GoogleDocID)
+		err := r.RejectSpec(ctx, spec, cleanupID)
+		if err != nil {
+			logger.Error("failed to reject spec", "error", err.Error())
+			r.FailedCount++
+		} else {
+			r.RejectedCount++
+		}
+	}
+
+	r.Logger.Info("completed stale spec rejection",
+		"total", len(specs),
+		"rejected", r.RejectedCount,
+		"failed", r.FailedCount)
+
+	return nil
+}
+
+// RejectSpec updates a specification document to rejected status
+func (r *RejectService) RejectSpec(
+	ctx context.Context,
+	spec *db.Spec,
+	cleanupID string,
+) error {
+	logger := r.Logger.With("spec_id", spec.ID, "doc_id", spec.GoogleDocID)
+
+	if r.Config.DryRun {
+		logger.Info("would reject spec (dry run)")
+		return nil
+	}
+
+	// Find the status cell coordinates
+	coords, err := r.findStatusCell(ctx, spec.GoogleDocID)
+	if err != nil {
+		return fmt.Errorf("failed to find status cell: %w", err)
+	}
+	if coords == nil {
+		return fmt.Errorf("document is not a draft/braindump")
+	}
+
+	// Update the Google Doc
+	if err := r.updateDocumentStatus(ctx, spec.GoogleDocID, coords, "Rejected"); err != nil {
+		return fmt.Errorf("failed to update document: %w", err)
+	}
+
+	updateData := map[string]any{
+		"status":    "Rejected",
+		"synced_at": time.Now(),
+	}
+	if err := r.DB.Model(spec).Where("id = ?", spec.ID).Updates(updateData).Error; err != nil {
+		return fmt.Errorf("failed to update spec status in database: %w", err)
```

Comment:
When returning an error from an exported function, you need to decide: do I want my callers to be able to see the details or type of a specific error returned by one of my dependencies? (in this case, gorm).

The answer is usually no: you don't want the users of `RejectService` to be able to know you are using gorm (that's an implementation detail). If you do decide to return gorm errors, you should usually document them in the Godoc, for example: `May return errors of type gorm.ErrRecordNotFound`.

That is because once you make something visible to your callers, it's very hard to hide it in the future.

So in most cases, you should usually opt for `%v` instead of `%w`.

I explain that in more detail here: https://alnvdl.github.io/2023/09/07/error-handling-in-go.html